File: Questions/Array/ContainsDuplicate/217-ContainsDuplicate.py
class Solution:
    def containsDuplicate1(self, nums: List[int]) -> bool:
        return len(set(nums)) != len(nums)

    def containsDuplicate2(nums: List[int]) -> bool:
        n = len(nums)
        if n > 1:
            nums.sort()
            for i in range(1, n):
                if nums[i-1] == nums[i]:
                    return True
        return False

    def containsDuplicate3(self, nums: List[int]) -> bool:
        hashset = set()
        for num in nums:
            if num in hashset:
                return True
            else:
                hashset.add(num)
        return False

File: Questions/Array/ContainsDuplicate/217-ContainsDuplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
     
        std::unordered_map<int, std::vector<int>> chainedHashBucket;
        
        for (int num : nums)
        {
            auto findResult = chainedHashBucket.find(num);
            if (findResult == chainedHashBucket.end())
            {
                chainedHashBucket.insert(std::make_pair(num, std::vector<int>{num}));
            }
            else
            {
                for (int sameHashNumber : findResult->second)
                    if (sameHashNumber == num) return true;
                
                findResult->second.push_back(num);
            }       
        }
        
        return false;
    }
};
